lib/archethic/utils.ex

```diff
@@ -811,4 +813,57 @@ defmodule Archethic.Utils do
       |> DateTime.from_naive!("Etc/UTC")
     end
   end
+
+  @doc """
+  Return the number of occurences for the cron job over the month
+
+  ## Examples
+
+      iex> Utils.number_of_reward_occurences_per_month("0 0 2 * * * *", ~N[2022-11-01 00:00:00.000000])
+      30
+
+      iex> Utils.number_of_reward_occurences_per_month("0 0 2 * * * *", ~N[2022-12-01 00:00:00.000000])
+      31
+
+      iex> Utils.number_of_reward_occurences_per_month("0 */5 * * * * *", ~N[2022-11-01 00:00:00.000000])
+      8640
+  """
+  @spec number_of_reward_occurences_per_month(String.t(), NaiveDateTime.t()) :: non_neg_integer()
+  def number_of_reward_occurences_per_month(
+        interval \\ Application.get_env(:archethic, RewardScheduler)[:interval],
+        current_datetime \\ NaiveDateTime.utc_now()
+      ) do
+    time = fn
+      true -> " 00:00:00Z"
+      false -> " 23:59:59Z"
+    end
+
+    date_to_datetime_converter = fn date, start_of_month? ->
+      time = time.(start_of_month?)
+
+      date
```

Comment:
I think you can directly use string interpolation: 
```elixir
NaiveDateTime.from_iso8601!("#{date}#{time}")
```

Reply:
thank you I will try it